src/controllers/vendor/vendorsController.js

```diff
@@ -0,0 +1,288 @@
+    const newVendor = await vendors.create({
+      fullName: req.body.fullName,
+      email: req.body.email,
+      password: hashedPassword,
+      phoneNumber: req.body.phone,
+      businessName: req.body.businessName,
+      businessAddress: req.body.businessAddress,
+      businessPhoneNumber: req.body.businessPhone,
+      businessEmail: req.body.businessEmail,
+      businessWebsite: req.body.businessWebsite,
+      businessDescription: req.body.businessDescription,
+      businessLogo: req.body.businessLogo,
+      productCategories: req.body.productCategories,
+      paymentMethods: req.body.paymentMethods,
+      status: req.body.status,
+    });
+    if (
+      !req.body.fullName
+      || !req.body.email
+      || !req.body.password
+      || !req.body.phoneNumber
+      || !req.body.businessName
+      || !req.body.businessAddress
+      || !req.body.businessPhoneNumber
+      || !req.body.businessEmail
+      || !req.body.businessWebsite
+      || !req.body.businessDescription
+      || !req.body.businessLogo
+      || !req.body.productCategories
+      || !req.body.paymentMethods
+      || !req.body.status
+    ) {
+      return res.json({ message: req.t('registerVendor_validation_001_msg') });
+    }
+    const transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        user: process.env.EMAIL_ADDRESS,
+        pass: process.env.EMAIL_PASSWORD,
+      },
+    });
+    const mailOptions = {
+      from: process.env.EMAIL_SENDER,
+      to: newVendor.email,
+      subject: 'Welcome to Cogito',
+      text: `Dear ${newVendor.fullName},\n\nWelcome to Cogito! We are excited to have you as a new vendor. Your business information has been successfully saved in our system.\n\nHere is the information you provided:\n\nFull Name: ${newVendor.fullName}\nBusiness Name: ${newVendor.businessName}\nBusiness Address: ${newVendor.businessAddress}\nBusiness Phone Number: ${newVendor.businessPhoneNumber}\nBusiness Email: ${newVendor.businessEmail}\nBusiness Website: ${newVendor.businessWebsite}\nBusiness Description: ${newVendor.businessDescription}\nProduct Categories: ${newVendor.productCategories}\nPayment Methods: ${newVendor.paymentMethods}\nRandom Password: ${req.body.password}, You can change this password only after login to our site.\n Login Here:\t http://localhost:9090/vendors/login \n\nThank you for choosing Cogito!\n\nBest regards,\nThe Cogito Team`,
+    };
+    transporter.sendMail(mailOptions, (mailError, info) => {
+      if (mailError) {
+        console.log(mailError);
+        res.status(500).json({
+          success: false,
+          message: req.t('registerVendor_500_001_msg'),
+          Error: mailError,
+        });
+      } else {
+        console.log(`Email sent: ${info.response}`);
+        res.status(201).json({
+          success: true,
+          message: req.t('registerVendor_201_msg'),
+          response: newVendor,
+        });
+      }
+    });
```

Comment:
How about this because a function that could be reusable anywhere ?

Reply:
Remember to remove console

Reply:
Have created a middleware in src/middleware/vendors/vendorSendMail.js with a sendMail function that will be called in my controller and passed with appropriate parameters to send email to the newly registered vendors. Also have removed all the console in my production code!!

Reply:
That nice